aea/test_tools/test_cases.py

```diff
@@ -220,21 +245,19 @@ def terminate_agents(
         :param signal: the signal for interuption
         :timeout: the timeout for interuption
         """
-        if subprocesses is None:
-            subprocesses = self.subprocesses
+        if not subprocesses:
```

Comment:
I suggest reverting so you don't need the type ignore and there is also nothing wrong with the previous code.